### PR TITLE
Security version number to container

### PR DESCRIPTION
**Pull request by lsubash**
Add svn field to container generation. SVN need
to be verified while doing container capsule
update. svn is added as end parameter to layout.

Format expected for container creation using '-cl' option for specifying components,
CMDL:Inp\cmdline.txt:$SVN

Signed-off-by: Subash Lakkimsetti <subash.lakkimsetti@intel.com>

### CODE REVIEW

File: BootloaderCorePkg/Tools/GenContainer.py [hunk 1]
```diff
@@ -803,6 +811,7 @@ def main():
                     'RSA3072_PKCS1_SHA2_384', 'RSA2048_PSS_SHA2_256', 'RSA3072_PSS_SHA2_384', 'NONE'], default='',  help='authentication algorithm')
     cmd_display.add_argument('-cd', dest='comp_dir', type=str, default='', help='Componet image input directory')
     cmd_display.add_argument('-td', dest='tool_dir', type=str, default='', help='Compression tool directory')
+    cmd_display.add_argument('-s', dest='svn', type=int, help='Security version number')
```
Comment (sagraw2) [inline BootloaderCorePkg/Tools/GenContainer.py:814]: help = 'Security version number for Container header'

File: BootloaderCorePkg/Tools/GenContainer.py [hunk 2]
```diff
@@ -822,6 +831,7 @@ def main():
     cmd_display.add_argument('-c',  dest='compress', choices=['lz4', 'lzma', 'dummy'], default='dummy', help='compression algorithm')
     cmd_display.add_argument('-k',  dest='key_file',  type=str, default='', help='Key Id or Private key file path to sign component')
     cmd_display.add_argument('-td', dest='tool_dir', type=str, default='', help='Compression tool directory')
+    cmd_display.add_argument('-s', dest='svn', type=int, required=True, help='Security version number')
```
Comment (sagraw2) [inline BootloaderCorePkg/Tools/GenContainer.py:834]: help = 'Security version number for Component'

File: BuildLoader.py [hunk 3]
```diff
@@ -1114,8 +1114,9 @@ def pre_build(self):
             if getattr(self._board, "GetKeyHashList", None):
                 key_hash_list = self._board.GetKeyHashList ()
 
+        svn = 0
```
Comment (sagraw2) [inline BuildLoader.py:1117]: Does this needs to be configured from Board config file ?

File: BootloaderCorePkg/Tools/CommonUtility.py [hunk 4]
```diff
@@ -360,6 +362,7 @@ def compress (in_file, alg, out_path = '', tool_dir = ''):
     compress_data = get_file_data(out_file)
     lz_hdr = LZ_HEADER ()
     lz_hdr.signature = sig.encode()
+    lz_hdr.svn = svn
```
Comment (sagraw2) [inline BootloaderCorePkg/Tools/CommonUtility.py:365]: What is the purpose of this svn ?

Comment (lsubash) [reply BootloaderCorePkg/Tools/CommonUtility.py:365]: This is referenced for components inside a container. Currently, There are no empty fields available in component headers. Components are compressed using  LZ compress and and appended to component header.
Eg: CompressHdr = (LOADER_COMPRESSED_HEADER *) CompBuf

Same "LOADER_COMPRESSED_HEADER" would be used for SBL stages compression also.

File: BuildLoader.py (hunk 3 above)
Comment (lsubash) [reply BuildLoader.py:1117]: Thanks. Yes we could get it from boardconfig files as new config param for KEYH container svn.

File: BootloaderCorePkg/Tools/CommonUtility.py (hunk 4 above)
Comment (sagraw2) [reply BootloaderCorePkg/Tools/CommonUtility.py:365]: Got it. Thanks.

**Pull request merged**